src/db/spanner/models.rs

```diff
@@ -783,6 +799,172 @@ impl SpannerDb {
+            "modified" => TypeCode::TIMESTAMP,
+        };
+
+        self.metrics
+            .clone()
+            .start_timer("storage.quota.update_existing_totals", None);
+        let calc_sql = if self.quota_enabled {
+            "SELECT SUM(BYTE_LENGTH(payload)), COUNT(*)
+            FROM bsos
+           WHERE fxa_uid = @fxa_uid
+             AND fxa_kid = @fxa_kid
+             AND collection_id = @collection_id
+           GROUP BY fxa_uid"
+        } else {
+            "SELECT COUNT(*)
+            FROM bsos
+           WHERE fxa_uid = @fxa_uid
+             AND fxa_kid = @fxa_kid
+             AND collection_id = @collection_id
+           GROUP BY fxa_uid"
+        };
+        let result = self
+            .sql(calc_sql)?
+            .params(params! {
+                "fxa_uid" => user.fxa_uid.clone(),
+                "fxa_kid" => user.fxa_kid.clone(),
+                "collection_id" => collection_id.to_string(),
+            })
+            .execute_async(&self.conn)?
+            .one_or_none()
+            .await?;
+        let set_sql = if let Some(result) = result {
+            // Update the user_collections table to reflect current numbers.
+            // If there are BSOs, there are user_collections (or else something
+            // really bad already happened.)
+            if self.quota_enabled {
+                let mut v = Value::new();
+                v.set_string_value(result[0].get_string_value().to_owned());
+                sqlparams.insert("total_bytes".to_owned(), v);
+                let mut v = Value::new();
+                v.set_string_value(result[1].get_string_value().to_owned());
+                sqlparams.insert("count".to_owned(), v);
+                sqltypes.insert(
+                    "total_bytes".to_owned(),
+                    crate::db::spanner::support::as_type(TypeCode::INT64),
+                );
+                sqltypes.insert(
+                    "count".to_owned(),
+                    crate::db::spanner::support::as_type(TypeCode::INT64),
+                );
+                "UPDATE user_collections
+                SET modified = @modified,
+                    count = @count,
+                    total_bytes = @total_bytes
+                WHERE fxa_uid = @fxa_uid
+                    AND fxa_kid = @fxa_kid
+                    AND collection_id = @collection_id"
+            } else {
+                "UPDATE user_collections
+                SET modified = @modified
+                WHERE fxa_uid = @fxa_uid
+                AND fxa_kid = @fxa_kid
+                AND collection_id = @collection_id"
+            }
+        } else {
+            // Otherwise, there are no BSOs that match, check to see if there are
+            // any collections.
+            let result = self
+                .sql(
+                    "SELECT 1 FROM user_collections
+                WHERE fxa_uid=@fxa_uid AND fxa_kid=@fxa_kid AND collection_id=@collection_id
+                LIMIT 1",
```

Comment:
we're directly querying the primary key (fxa_uid/kid/collection_id) so the LIMIT 1's superfluous
```suggestion
                WHERE fxa_uid=@fxa_uid AND fxa_kid=@fxa_kid AND collection_id=@collection_id",
```